code/controllers/subsystems/mapping.dm

```diff
@@ -32,6 +32,17 @@ SUBSYSTEM_DEF(mapping)
 		map_templates[T.name] = T
 	preloadBlacklistableTemplates()
 
+/datum/controller/subsystem/mapping/proc/includeTemplate(var/datum/map_template/map_template, var/list/banned_maps)
+	if(!initial(map_template.id))
+		return FALSE
+	var/datum/map_template/MT = new map_template()
+	if(banned_maps)
+		for(var/mappath in MT.mappaths)
+			if(banned_maps.Find(mappath))
+				return FALSE
+	map_templates[MT.name] = MT
+	. = MT
```

Comment:
This proc is mixing returning reference and value types (a template or `FALSE`)
Should either return instance/`null` or `TRUE`/`FALSE`

Reply:
Line 43-44 should probable be replaced with `return TRUE`

https://github.com/NebulaSS13/Nebula/pull/22/files#diff-c2e5790416598e06d4b9215742ea8c70R58-R59 should be replaced with
```
if(includeTemplate(item, banned_maps)
	map_templates[MT.name] = MT
	. = MT
else
	continue
```
That way https://github.com/NebulaSS13/Nebula/pull/22/files#diff-de1774595a6a972d2ecda2c7215a5aacR83-R86 can simply be changed to
```
/datum/controller/subsystem/mapping/includeTemplate(var/datum/map_template/map_template, var/banned_maps)
	return TRUE
```